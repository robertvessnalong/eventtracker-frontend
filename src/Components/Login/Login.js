import { useState, useContext } from 'react';
import { useHistory } from 'react-router-dom';
import { Form, Button, Container, Alert } from 'react-bootstrap';
import { Redirect } from 'react-router';
import UserContext from '../../Context/UserContext';

const Login = () => {
  const history = useHistory();
  const { login, loggedIn } = useContext(UserContext);
  const formInitalState = {
    email: '',
    password: '',
  };
  const [formData, setFormData] = useState(formInitalState);
  const [error, setError] = useState(false);
  const [errorMsg, setErrorMsg] = useState([]);

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFormData((data) => ({
      ...data,
      [name]: value,
    }));
  };

  const handleSubmit = async (e) => {
    e.preventDefault();
    const res = await login(formData);
    if (res.error) {
      setError(true);
      setErrorMsg(res.error);
    } else if (res.login) {
      setFormData(formInitalState);
      history.push('/profile');
    }
  };

  if (loggedIn) {
    return <Redirect to='/profile'></Redirect>;
  }

  return (
    <>
      <div className='Login Center'>
        <Container>
          <Form onSubmit={handleSubmit}>
            {error && <Alert variant='danger'>{errorMsg}</Alert>}
            <Form.Group className='mb-3' controlId='loginEmail'>
              <Form.Label>Email address</Form.Label>
              <Form.Control
                type='email'
                placeholder='Enter email'
                onChange={handleChange}
                name='email'
              />
              <Form.Text className='text-muted'>
                We'll never share your email with anyone else.
              </Form.Text>
            </Form.Group>
            <Form.Group className='mb-3' controlId='loginPassword'>
              <Form.Label>Password</Form.Label>
              <Form.Control
                type='password'
                placeholder='Password'
                onChange={handleChange}
                name='password'
              />
            </Form.Group>
            <Button className='blue-btn' type='submit'>
              Submit
            </Button>
          </Form>
        </Container>
      </div>
    </>
  );
};

export default Login;
