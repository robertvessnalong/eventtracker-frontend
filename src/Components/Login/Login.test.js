import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Login from './Login';
import TestUser from '../../Testing/TestUser';
import UserContext from '../../Context/UserContext';


describe('Events Component', () => {
  // Smoke Test
  test('renders without crashing', async () => {
    render(
      <UserContext.Provider value={{ user: TestUser }}>
        <Login />
      </UserContext.Provider>
    );

    const label = screen.getByLabelText(/Email Address/i);
    expect(label).toBeInTheDocument();
  });

  // Snapshot
  test('matches snapshot', async () => {
    const { asFragment } = render(
      <UserContext.Provider value={{ user: TestUser }}>
        <Login />
      </UserContext.Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
