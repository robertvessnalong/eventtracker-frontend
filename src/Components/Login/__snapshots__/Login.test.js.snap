// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Login Center"
  >
    <div
      class="container"
    >
      <form
        class=""
      >
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="loginEmail"
          >
            Email address
          </label>
          <input
            class="form-control"
            id="loginEmail"
            name="email"
            placeholder="Enter email"
            type="email"
          />
          <small
            class="text-muted form-text"
          >
            We'll never share your email with anyone else.
          </small>
        </div>
        <div
          class="mb-3"
        >
          <label
            class="form-label"
            for="loginPassword"
          >
            Password
          </label>
          <input
            class="form-control"
            id="loginPassword"
            name="password"
            placeholder="Password"
            type="password"
          />
        </div>
        <button
          class="blue-btn btn btn-primary"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
